app/Http/Controllers/ShiftController.php

```diff
@@ -0,0 +1,414 @@
+
+class ShiftController extends Controller
+{
+    /**
+     * Display the specified resource.
+     * Returns JSON-formated contents of a schedule entry.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $shift = Shift::where('id', '=', $id)
+            ->with('type', 'getPerson.getClub')
+            ->firstOrFail();
+
+        // Person NULL means "=FREI=" - check for it every time you query a relationship
+        $ldapId = !is_null($shift->getPerson) ? $shift->getPerson->prsn_ldap_id : "";
+        $name = !is_null($shift->getPerson) ? $shift->getPerson->prsn_name : "=FREI=";
+        $status = !is_null($shift->getPerson) ? $shift->getPerson->prsn_status : "";
+        $clubTitle = !is_null($shift->getPerson) ? $shift->getPerson->getClub->clb_title : "";
+        $isCurrentUser = $ldapId == Session::get('userId');
+        $response = [
+            'id'                => $shift->id,
+            'jbtyp_title'       => $shift->type->title(),
+            'prsn_name'         => $name,
+            'prsn_ldap_id'      => $ldapId,
+            'prsn_status'       => $status,
+            'clb_title'         => $clubTitle,
+            'comment'=> $shift->comment,
+            'start'  => $shift->start,
+            'end'    => $shift->end,
+            'updated_at'        => $shift->updated_at,
+            'is_current_user'   => $isCurrentUser
+        ];
+
+        return response()->json($response);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     * Changes contents of the entry specified by ID to contents in the REQUEST
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        // Check if it's our form (CSRF protection)
+        if ( Session::token() !== Input::get( '_token' ) ) {
+            return response()->json('Fehler: die Session ist abgelaufen. Bitte aktualisiere die Seite und logge dich ggf. erneut ein.', 401);
+        }
+
+        Utilities::clearIcalCache();
+
+        // If we only want to modify the jobtype via management pages - do it without evaluating the rest
+        if ( !empty($request->get('jobtypeId')) && is_numeric($request->get('jobtypeId')) ) {
+
+            // Find the corresponding shift object
+            $shift = Shift::where('id', '=', $request->get('entryId'))
+                ->first();
+
+            // Substitute values
+            $shift->shifttype_id = $request->get('jobtypeId');
+            $shift->save();
+
+            // Log changes
+            ScheduleController::logAction($shift, "Diensttyp geändert");
+
+            // Formulate the response
+            return response()->json([
+                "entryId" => $shift->id,
+                "updatedJobtypeTitle" => ShiftType::where('id', '=', $shift->shifttype_id)->first()->jbtyp_title
```

Comment:
Will do, but on a separate branch after this PR is merged.  This PR is already too huge.